### PR TITLE
[Monitor] Fixing incidents #3585 [Monitor] Breaking change found in AutoScale spec (Swagger spec) and #3293 [Monitor] Add serviceBusRuleId to the DiagnosticSettings resource (Swagger spec)

### PR DESCRIPTION
**Pull request by gucalder**
## Description

Generating code with the latest Swagger spec

Fixing two incidents:
#3585 [Monitor] Breaking change found in AutoScale spec (Swagger spec) and 
#3293 [Monitor] Add serviceBusRuleId to the DiagnosticSettings resource (Swagger spec)

---

This checklist is used to make sure that common guidelines for a pull request are followed.
- [x] Please add REST spec PR link to the SDK PR
- [x] **I have read the [contribution guidelines](https://github.com/Azure/azure-sdk-for-net/blob/AutoRest/.github/CONTRIBUTING.md).**
- [x] **The pull request does not introduce [breaking changes](https://github.com/Azure/azure-sdk-for-net/blob/AutoRest/Documentation/breaking-changes.md).**

### [General Guidelines](https://github.com/Azure/azure-sdk-for-net/blob/AutoRest/.github/CONTRIBUTING.md#general-guidelines)
- [x] Title of the pull request is clear and informative.
- [x] There are a small number of commits, each of which have an informative message. This means that previously merged commits do not appear in the history of the PR. For more information on cleaning up the commits in your PR, [see this page](https://github.com/Azure/azure-powershell/blob/dev/documentation/cleaning-up-commits.md).

### [Testing Guidelines](https://github.com/Azure/azure-sdk-for-net/blob/AutoRest/.github/CONTRIBUTING.md#testing-guidelines)
- [x] Pull request includes test coverage for the included changes.

### [SDK Generation Guidelines](https://github.com/Azure/azure-sdk-for-net/blob/AutoRest/.github/CONTRIBUTING.md#sdk-generation-guidelines)
- [x] If an SDK is being regenerated based on a new swagger spec, a link to the pull request containing these swagger spec changes has been included above.
- [x] The generate.cmd file for the SDK has been updated with the version of AutoRest, as well as the commitid of your swagger spec or link to the swagger spec, used to generate the code.
- [x] The `*.csproj` and `AssemblyInfo.cs` files have been updated with the new version of the SDK.

### CODE REVIEW

File: src/SDKs/Monitor/Monitor.Tests/Monitor.Tests.csproj [hunk 1]
```diff
@@ -6,9 +6,9 @@
 		<PackageId>Monitor.Tests</PackageId>
 		<VersionPrefix>1.0.0-preview</VersionPrefix>
 	</PropertyGroup>
-	<!--<PropertyGroup>
-		<TargetFrameworks>netcoreapp1.1</TargetFrameworks>
-	</PropertyGroup>-->
+	<PropertyGroup>
```
Comment (dsgouda) [inline src/SDKs/Monitor/Monitor.Tests/Monitor.Tests.csproj]: Please do not un-comment these lines. We have moved to netcoreapp2.0 and will fix these for all csprojs at a later time

File: src/SDKs/Monitor/Management.Monitor/Properties/AssemblyInfo.cs [hunk 2]
```diff
@@ -7,8 +7,8 @@
 [assembly: AssemblyTitle("Microsoft Azure Monitor Library")]
 [assembly: AssemblyDescription("Provides Microsoft Azure Monitor operations.")]
 
-[assembly: AssemblyVersion("0.20.0.0")]
-[assembly: AssemblyFileVersion("0.20.0.0")]
+[assembly: AssemblyVersion("0.20.1.0")]
```
Comment (dsgouda) [inline src/SDKs/Monitor/Management.Monitor/Properties/AssemblyInfo.cs]: `AssemblyVersion` must always be the first major release version i.e. (0.9.0), only `AssemblyFileVersion` needs to be updated for non-major version updates

**Review by dsgouda — CHANGES_REQUESTED** [review]
@gucalder looks like the latest REST API version is now stable [here](https://github.com/Azure/azure-rest-api-specs/blob/master/specification/monitor/resource-manager/readme.md#tag-package-2018-03)
We recommend releasing a stable Nuget package for stable REST API versions. Do you have any plans to release a stable version?

**Review by dsgouda — CHANGES_REQUESTED** [review]
Looks great for the most part, please merge all the commits into a single commit and we should be good to merge

File: src/SDKs/Monitor/Management.Monitor/Properties/AssemblyInfo.cs (hunk 2 above)
Comment (dsgouda) [reply src/SDKs/Monitor/Management.Monitor/Properties/AssemblyInfo.cs]: @gucalder I missed that this is 0.x.x version, in that case your earlier change was correct. Apologies for the incorrect feedback.
Please revert both `AssemblyVersion` and `AssemblyFileVersion` to `0.20.1.0`

**Comment by gucalder** [comment]
I honestly do not know how to change the history of my remote repo to make this squash possible. In my local repo, no problem, but there, hmmm.
Isn't it easier to do when the PR is merged?

**Comment by dsgouda** [comment]
Unfortunately, there is no easy way. Copy over your changes to a temp local location. Recreate a branch with the same name (you might want to rename the existing branch to preserve changes) and then commit and force push
As an aside, the recommended "git way" of working on a repo is to create branches for every change you wish to make. Hope this helps

**Review by dsgouda — APPROVED** [review]
LGTM

**Comment by dsgouda** [comment]
Will merge on commits being squashed

**Comment by gucalder** [comment]
Changes squashed to single commit.

**Pull request merged**